src/main/java/de/tum/in/www1/artemis/service/plagiarism/TextPlagiarismDetectionService.java

```diff
@@ -61,13 +62,10 @@ public TextPlagiarismDetectionService(TextSubmissionExportService textSubmission
      * @return List containing the latest text submission for every participation
      */
     public List<TextSubmission> textSubmissionsForComparison(TextExercise exerciseWithParticipationsAndSubmissions, int minimumScore, int minimumSize) {
-        var textSubmissions = exerciseWithParticipationsAndSubmissions.getStudentParticipations().parallelStream().filter(participation -> participation.getStudent().isPresent())
-                .filter(participation -> !authCheckService.isAtLeastTeachingAssistantForExercise(exerciseWithParticipationsAndSubmissions, participation.getStudent().get()))
+        var textSubmissions = exerciseWithParticipationsAndSubmissions.getStudentParticipations().parallelStream().filter(plagiarismService.filterForStudents())
                 .map(Participation::findLatestSubmission).filter(Optional::isPresent).map(Optional::get).filter(submission -> submission instanceof TextSubmission)
```

Comment:
You made the decision to extract the former lines 64 to 69 into the methods In PlagiarismService.filterParticipationMinimumScore() and PlagiarismService.hasMinimumScore(). Why did you decide against doing the same with the minimum size?

Reply:
Having those as two separate methods makes it much more readable for developers, as conditions are explicitly named

Reply:
@jakubriegel The question was less about PlagiarismService.filterParticipationMinimumScore() and PlagiarismService.hasMinimumScore() being two separate methods. The question was more: Why is there no PlagiarismService.filterParticipationMinimumSize() and PlagiarismService.hasMinimumSize()?

Reply:
with minimum size it's a bit more tricky, because this depends on the submission type. The logic is different, so extracting it, is not easy. One way would be to add an abstract method in Submission and implement it in the subclasses, however this would not work for programming submissions. Therefore, this is out of scope for now